qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -249,7 +249,11 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
                 )
 
             all_circuit_results.append(circuit_results)
-            oracle_evaluation = amplification_problem.is_good_state(top_measurement)
+            if isinstance(amplification_problem.oracle, PhaseOracle):
+                oracle_evaluation = amplification_problem.is_good_state(top_measurement[::-1])
```

Comment:
This seems a bit volatile. If we come up with new oracles we'll potentially have to add more distinctions here. Can't we make `PhaseOracle` use the same order as Grover? Or should we change the order assumed in Grover?

Reply:
I agree with that. Actually, I was thinking to change `PhaseOracle` at first. 
What do you think? @boschmitt If you're ok, I'm thinking to change the bit order for evaluation in `PaseOracle`.

Reply:
Sure you can change `PhaseOracle`. I still think it's not a good long-term solution, but works well as a quick fix

Reply:
Thx! I will change the code in `PhaseOracle` for now.

Reply:
@a-matsuo Did you have time to move the change to `PhaseOracle`? 🙂

Reply:
@Cryoris @woodsp-ibm Sorry, I forgot about this PR. I will work on this!